### PR TITLE
feat: [PLG-2628]: Added delete credit card endpoint

### PR DESCRIPTION
**Pull request by thomas-grasman**
## Describe your changes
Added an endpoint for deleting credit cards from local mongo.
Only harness support users are allowed to use this endpoint.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)

### CODE REVIEW

**Comment by thomas-grasman** [comment]
trigger messagecheck

**Comment by thomas-grasman** [comment]
trigger uts

**Review by neelam-harness — CHANGES_REQUESTED** [review]

**Review by kapil-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: Some of the checks are too old. 5 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java` line 85 at r1](https://reviewable.io/reviews/harness/harness-core/51586#-NbxTIpH48emeoRbC5nY:-NbxTIpH48emeoRbC5nZ:boj8g2b) ([raw file](https://github.com/harness/harness-core/blob/04618c985701458fccf67432d02e68cc11623210/930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java#L85)):*
> ```Java
>   }
> 
>   @DELETE
> ```

Delete APIs should have identifier of the resource to be deleted as a path param and not as a query param 
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java` line 91 at r1](https://reviewable.io/reviews/harness/harness-core/51586#-NbxTEFe2ImYEkFOwP8M:-NbxTEFfA3jGaPiVGTaR:bw0f0k3) ([raw file](https://github.com/harness/harness-core/blob/04618c985701458fccf67432d02e68cc11623210/930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java#L91)):*
> ```Java
>           @Parameter(required = true, description = ACCOUNT_PARAM_MESSAGE) @NotNull @QueryParam(
>                   NGCommonEntityConstants.ACCOUNT_KEY) @AccountIdentifier String accountIdentifier,
>           @Valid @RequestBody(description = "Non-sensitive credit card information") @NotNull CreditCardDTO creditCardDTO) {
> ```

a full DTO is not required and only creditCardId would be sufficient
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 109 at r1](https://reviewable.io/reviews/harness/harness-core/51586#-NbxTsRK0PBP6-OOi_kx:-NbxTsRK0PBP6-OOi_ky:b-bsemwl) ([raw file](https://github.com/harness/harness-core/blob/04618c985701458fccf67432d02e68cc11623210/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L109)):*
> ```Java
>   public CreditCardResponse deleteCreditCard(CreditCardDTO creditCardDTO) {
>     boolean isHarnessSupportEnabled = CGRestUtils.getResponse(
>         accountClient.checkIfHarnessSupportEnabledForAccount(creditCardDTO.getAccountIdentifier()));
> ```

I think you want to check whether the user who wants to delete a CC is part of Harness support user group or not.
Then why are you checking it for an account?

right implementation is here: software.wings.resources.AccountResource#createNgLicense

**Review by thomas-grasman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: 6 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java` line 85 at r1](https://reviewable.io/reviews/harness/harness-core/51586#-NbxTIpH48emeoRbC5nY:-Nbzp0KMDXgSvgwWrIQm:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/04618c985701458fccf67432d02e68cc11623210/930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java#L85)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

Delete APIs should have identifier of the resource to be deleted as a path param and not as a query param
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java` line 91 at r1](https://reviewable.io/reviews/harness/harness-core/51586#-NbxTEFe2ImYEkFOwP8M:-Nbzp13v3z6SuBKy9dTC:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/04618c985701458fccf67432d02e68cc11623210/930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java#L91)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

a full DTO is not required and only creditCardId would be sufficient
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 109 at r1](https://reviewable.io/reviews/harness/harness-core/51586#-NbxTsRK0PBP6-OOi_kx:-Nbzp3Ux2I-J8XfLoo3G:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/04618c985701458fccf67432d02e68cc11623210/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L109)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

I think you want to check whether the user who wants to delete a CC is part of Harness support user group or not.
Then why are you checking it for an account?

right implementation is here: software.wings.resources.AccountResource#createNgLicense
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 113 at r3](https://reviewable.io/reviews/harness/harness-core/51586#gh-1295241472:-Nbzp4zi6NiqjfEs6EQ4:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L113)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

What happens here and after this if the creditCard is not found ?
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java` line 109 at r3](https://reviewable.io/reviews/harness/harness-core/51586#gh-1295242292:-Nbzp6ML9Emq4bpDkI2A:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java#L109)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

Let's add a unit test to cover the case when a credit card is not found in the database but the api for deleteCard is being called ?
</blockquote></details>

Done.

**Comment by thomas-grasman** [comment]
trigger uts

File: 930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java [hunk 1]
```diff
@@ -98,6 +105,26 @@ public CreditCardResponse saveCreditCard(CreditCardDTO creditCardRequest) {
...
+  @Override
+  public CreditCardResponse deleteCreditCard(String accountIdentifier, String creditCardIdentifier) {
+    SourcePrincipalContextData sourcePrincipalContextData =
+        GlobalContextManager.get(SourcePrincipalContextData.SOURCE_PRINCIPAL);
+    String userId = sourcePrincipalContextData.getPrincipal().getName();
+
+    boolean isHarnessSupportEnabled = CGRestUtils.getResponse(accountClient.isHarnessSupportUserId(userId));
+
+    if (!isHarnessSupportEnabled) {
+      throw new NotAuthorizedException("Only Harness support users can delete credit cards.");
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java:117]: Let's log the error and then throw.

File: 930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java [hunk 2]
```diff
@@ -98,6 +105,26 @@ public CreditCardResponse saveCreditCard(CreditCardDTO creditCardRequest) {
...
+    }
+    CreditCard creditCard = creditCardRepository.findByCreditCardIdentifier(creditCardIdentifier);
+    if (creditCard == null) {
+      throw new InvalidArgumentsException(
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java:121]: Let's the log the error

**Review by neelam-harness — CHANGES_REQUESTED** [review]

**Review by neelam-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: 6 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 113 at r3](https://reviewable.io/reviews/harness/harness-core/51586#gh-1295241472:-Nc-5ZPgBagqm_XY3u2n:b-1k5g2) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L113)):*
<details><summary><i>Previously, thomas-grasman (Thomas Grasman) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

This is a question :) Please help clarify the use case  
___
*[`930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/51586#-Nc-5p1_82BtU4EKyywG:-Nc-5p1_82BtU4EKyywH:b-o44pos) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java#L106)):*
> ```Java
>   }
> 
>   @Test()
> ```

Does this still run the Test if you have the annotation as 
`@Test()` ?
___
*[`930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java` line 109 at r3](https://reviewable.io/reviews/harness/harness-core/51586#gh-1295242292:-Nc-5yab6bQCawo8Vhrq:b-imdyd) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java#L109)):*
<details><summary><i>Previously, thomas-grasman (Thomas Grasman) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Is there a new commit with the unit test requested above? In this diff I don't see the new unit test this seems to be the same unit test.

**Review by thomas-grasman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: 6 file(s) to review. 2 change requests. 8. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 113 at r3](https://reviewable.io/reviews/harness/harness-core/51586#gh-1295241472:-Nc-6jnL4MDpvgrM0l-_:bixit3a) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L113)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

This is a question :) Please help clarify the use case
</blockquote></details>

It cannot be deleted if it is not found.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 117 at r4](https://reviewable.io/reviews/harness/harness-core/51586#gh-1296388552:-Nc-9CCd4hPDehV0B_L9:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9a4e83850c4cc14aa6c67d14ade9d0a0b703d7ff/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L117)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

Let's log the error and then throw.
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 121 at r4](https://reviewable.io/reviews/harness/harness-core/51586#gh-1296388737:-Nc-9EjE9WuJbMEYMTVk:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9a4e83850c4cc14aa6c67d14ade9d0a0b703d7ff/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L121)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

Let's the log the error
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/51586#-Nc-5p1_82BtU4EKyywG:-Nc-9IEEBMJytCaUQme3:bor45ae) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java#L106)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

Does this still run the Test if you have the annotation as 
`@Test()` ?
</blockquote></details>

Yes, but I removed the parenthesis anyway.
___
*[`930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java` line 109 at r3](https://reviewable.io/reviews/harness/harness-core/51586#gh-1295242292:-Nc-9Q6q5UuzfG17UNMj:b8stt8f) ([raw file](https://github.com/harness/harness-core/blob/04ad11aa7858bb14297dc8c40fbfa3b2d32f0451/930-ng-license-manager/src/test/java/io/harness/subscription/services/CreditCardServiceImplTest.java#L109)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

Is there a new commit with the unit test requested above? In this diff I don't see the new unit test this seems to be the same unit test.
</blockquote></details>

This case is covered by testDeleteCardExpectsInvalidArgumentsException

**Comment by thomas-grasman** [comment]
trigger uts

**Review by kapil-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: 6 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java` line 87 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc17eqk5vtO1u6eS5az:-Nc17eqk5vtO1u6eS5b-:b-frzu32) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java#L87)):*
> ```Java
> 
>   @DELETE
>   @Path("{creditCardIdentifier}")
> ```

creditCardIdentifier is a very big path param, not required, just shorten it to "identifier"
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 111 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc1Bg20CNqFkd3mz_6T:-Nc1Bg24CpQW1r9n3UWg:bswx54d) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L111)):*
> ```Java
>   public CreditCardResponse deleteCreditCard(String accountIdentifier, String creditCardIdentifier) {
>     SourcePrincipalContextData sourcePrincipalContextData =
>         GlobalContextManager.get(SourcePrincipalContextData.SOURCE_PRINCIPAL);
> ```

are you sure that this is the way to get a user's context? Is this tested in local?
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 114 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc1COL9APQRk3dHueiw:-Nc1COLACRX-JBO03ABh:bibnfoc) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L114)):*
> ```Java
>     String userId = sourcePrincipalContextData.getPrincipal().getName();
> 
>     boolean isHarnessSupportEnabled = CGRestUtils.getResponse(accountClient.isHarnessSupportUserId(userId));
> ```

all client calls should be in a try catch clause and more importantly I am not able to find isHarnessSupportUserId() method anywhere in the codebase :( 
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 121 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc1EEKOCZzOS-zgdoMi:-Nc1EEKP5ll_Vzx391yr:b-cksq3a) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L121)):*
> ```Java
>           userId);
>       log.error(errorMessage);
>       throw new NotAuthorizedException(errorMessage);
> ```

lets use UnauthorizedException which is part of our in house WingsException



_Code snippet:_
```Java
software.wings.resources.FeatureFlagResource#getGloballyEnabledFlags
```

**Review by thomas-grasman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: 6 file(s) to review. 2 change requests. 9. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java` line 87 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc17eqk5vtO1u6eS5az:-Nc1NX6413kF0VeebNQM:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/resource/CreditCardResource.java#L87)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

creditCardIdentifier is a very big path param, not required, just shorten it to "identifier"
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 111 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc1Bg20CNqFkd3mz_6T:-Nc1NY6G4cgPT_KlyXro:b-2xto48) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L111)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

are you sure that this is the way to get a user's context? Is this tested in local?
</blockquote></details>

Yes. Here is a debug image:


___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 114 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc1COL9APQRk3dHueiw:-Nc1NkUXDcIIHGQNLLUI:b-skall7) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L114)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

all client calls should be in a try catch clause and more importantly I am not able to find isHarnessSupportUserId() method anywhere in the codebase :(
</blockquote></details>

Added try/catch. isHarnessSupportUserId is part of io.harness.account.AccountClient in 945-account-mgmt
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 121 at r5](https://reviewable.io/reviews/harness/harness-core/51586#-Nc1EEKOCZzOS-zgdoMi:-Nc1Oz4w5njPw0ky2Hl3:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2fde555eb0ddd2dce25aa20109cbd5f19c8d3a1f/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L121)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

lets use UnauthorizedException which is part of our in house WingsException
</blockquote></details>

Done.

**Review by kapil-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: 6 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java` line 118 at r7](https://reviewable.io/reviews/harness/harness-core/51586#-Nc1Wpwa2EYWiEI9F3Aa:-Nc1WpwoAt4BB4i8u38x:b-ko261b) ([raw file](https://github.com/harness/harness-core/blob/bb420b42789ab2920646b4ea248d0a74bbd2e9f2/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/CreditCardServiceImpl.java#L118)):*
> ```Java
>       isHarnessSupportEnabled = CGRestUtils.getResponse(accountClient.isHarnessSupportUserId(userId));
>     } catch (Exception e) {
>       log.error(e.getMessage(), e);
> ```

1st argument of log.error() should be a message from use and not exception itself, e.g. 



_Code snippet:_
```Java
log.error("client call to cg-manager failed due to: ", e);
```

**Comment by thomas-grasman** [comment]
trigger uts

**Review by kapil-harness — COMMENTED** [review]
> Reviewed 1 of 5 files at r1, 2 of 5 files at r4, 1 of 2 files at r7, 2 of 2 files at r8.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51586)* status: 6 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained.

**Comment by neelam-harness** [comment]
trigger uts

**Comment by thomas-grasman** [comment]
trigger review

**Pull request merged**